cumulus/pallets/parachain-system/src/lib.rs

```diff
@@ -699,6 +699,15 @@ pub mod pallet {
 			let post = frame_system::Pallet::<T>::do_apply_authorize_upgrade(code)?;
 			Ok(post)
 		}
+
+		/// Set the claim queue offset.
+		#[pallet::call_index(4)]
+		#[pallet::weight((T::WeightInfo::set_claim_queue_offset(), DispatchClass::Operational))]
+		pub fn set_claim_queue_offset(origin: OriginFor<T>, value: u8) -> DispatchResult {
```

Comment:
I think we can and should leave this a bit more generic. Instead of making this a storage value, we could have it configurable via a trait in the pallet config:

```rust
trait ClaimQueueOffsetStrategy {
fn claim_queue_offset() -> u8;
}
```

This way, we don't need any storage for the constant case and for dynamic cases the user runtime can base it directly on whatever logic they have. E.g. a consensus like [this one](https://github.com/paritytech/polkadot-sdk/issues/4813) could provide a callback dynamically setting the offset, according to current block builder rules.

Reply:
Then we also get the configurable "default" configuration for free.

Reply:
I like this idea, but it would be a breaking change for the parachain builders as they need to do this additional configuration in the runtime ?

Reply:
Depends. If we can provide a default implementation that works in all cases we are good. E.g. determine whether it should be 1 or 0 based on async backing settings.

Otherwise ... looking at the diff of this PR, this is already a breaking change, in the sense that people need to adjust their code.

Reply:
Worth mentioning, the provided implementation could also read from a passed in inherent, if the parachain wants to give (some) control to the block producer.

Reply:
With this interface we could also make it compatible for the lookahead. Maybe we should change this in general to:

```
trait DetermineCoreIndex {
    fn core_index() -> { offset, selector };
}
```

(Naming is shit I know)

However, for packing multiple blocks in one PoV this will for sure need to be made generic any way. This interface could bring us the freedom.

Reply:
Good idea @eskimor

Reply:
Good suggestion! I implemented this. Have a look!